src/kernel/kernel.ts

```diff
@@ -1058,25 +1058,32 @@ export class Syscalls {
 	open(task: ITask, path: string, flags: string, mode: number, cb: (err: number, fd: number) => void): void {
 		let fullpath = join(task.cwd, path);
 		// FIXME: support CLOEXEC
-		this.kernel.fs.open(fullpath, flags, mode, (err: any, fd: any) => {
-			let f: IFile;
-			if (err && err.errno === EISDIR) {
-				// TODO: update BrowserFS to open() dirs
-				f = new DirFile(this.kernel, fullpath);
-			} else if (!err) {
-				f = new RegularFile(this.kernel, fd);
-			} else {
-				if (typeof err === 'number')
-					cb(err, -1);
-				else if (err && err.errno)
-					cb(-err.errno, -1);
-				else
-					cb(-1, -1);
-				return;
-			}
-			let n = task.addFile(f);
-			cb(0, n);
-		});
+
+		let f: IFile;
+
+		if (fullpath === "/dev/null" ) {
+			f = new NullFile();
```

Comment:
almost -- best to put
```
		let n = task.addFile(f);
		cb(0, n);
		return;
```
inside this if, and return early, and then just leave the rest of the function as-is (not inside an else).  the two code-paths can't share the call to `task.addFile(f)`, because the call to kernel.fs.open may be delayed (and so addFile needs to be called in the callback).